src/pkg/cli/client/byoc/do/byoc.go

```diff
@@ -644,13 +644,32 @@ func (b *ByocDo) update(service composeTypes.ServiceConfig) *defangv1.ServiceInf
 	return si
 }
 
-func (b *ByocDo) setUp(ctx context.Context) error {
+func (b *ByocDo) needsSetup(ctx context.Context) (bool, error) {
 	projectCdImageTag, err := b.getCdImageTag(ctx)
+
 	if err != nil {
-		return err
+		if strings.Contains(err.Error(), "no bucket found") {
+			return true, nil
+		} else {
+			return true, err
+		}
 	}
 
+	b.cdImageTag = projectCdImageTag
 	if b.SetupDone && b.cdImageTag == projectCdImageTag {
```

Comment:
This. We don't need this `==`

Reply:
OK. How does this look then? ca0a6496e1215525c6538e7f4797f3c4dc501425

Reply:
That looks good!